In [74]:
import seaborn as sns
tips=sns.load_dataset("tips")
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
print(tips)


     total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]


In [75]:
tips.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


In [76]:
tips.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

The average tip amount is $ 2.998279




The average bill amount is $ 19.785943

The average bill percentage is 15.16 %

There are 157 males and 87 females customers

The higest tip is on Sat	with $ 260.4

Among lunch and Dinner,the highest tip is for dinner with $546.07

The tip on week days is $ 223.79

The tip on weekends in $507.78999999999996

Saturday is the most busiest day with 87 entries

Male tiped most with average 	$ 3.089618

Women tip more during dinner 52

There seems a direct correleation between the size and tips(almost like y=x)

There is no difference between smokers and non smokers in tipping

There is no difference between female and male smokers in tipping









In [77]:
total_tip=tips["tip"].sum()
print(total_tip)
total_bill=tips["total_bill"].sum()
print(total_bill)
avg_tip_percentage=(total_tip/total_bill)*100
print(avg_tip_percentage)


731.5799999999999
4827.77
15.153580224410025


In [78]:
tips.groupby("day")["tip"].sum().sort_values(ascending=False)


<ipython-input-78-3790118739>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("day")["tip"].sum().sort_values(ascending=False)


day
Sat     260.40
Sun     247.39
Thur    171.83
Fri      51.96
Name: tip, dtype: float64

In [79]:
tips.groupby("time")["tip"].sum().sort_values(ascending=False)

<ipython-input-79-1985606639>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("time")["tip"].sum().sort_values(ascending=False)


time
Dinner    546.07
Lunch     185.51
Name: tip, dtype: float64

In [80]:
tips.nunique()
#let check if they want to tip whole number or decimal number

total_bill    229
tip           123
sex             2
smoker          2
day             4
time            2
size            6
dtype: int64

In [81]:
tips.groupby("sex").count()

<ipython-input-81-294839457>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("sex").count()


total_bill  tip  smoker  day  time  size
sex                                             
Male           157  157     157  157   157   157
Female          87   87      87   87    87    87

In [82]:
tips.groupby("sex")["tip"].mean()

<ipython-input-82-1777249021>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("sex")["tip"].mean()


sex
Male      3.089618
Female    2.833448
Name: tip, dtype: float64

In [83]:
tips.groupby("smoker")["tip"].mean()

<ipython-input-83-637511320>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("smoker")["tip"].mean()


smoker
Yes    3.008710
No     2.991854
Name: tip, dtype: float64

In [84]:
week_day=tips.groupby("day")["tip"].sum()
print(week_day)
sum_on_week_days=week_day["Thur"]+week_day["Fri"]
sum_on_weekends=week_day["Sat"]+week_day["Sun"]
print(sum_on_week_days)
print(sum_on_weekends)


day
Thur    171.83
Fri      51.96
Sat     260.40
Sun     247.39
Name: tip, dtype: float64
223.79
507.78999999999996


<ipython-input-84-692611131>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  week_day=tips.groupby("day")["tip"].sum()


In [85]:
tips.groupby("day").count().sort_values(by="tip",ascending=False)

<ipython-input-85-1488306508>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("day").count().sort_values(by="tip",ascending=False)


total_bill  tip  sex  smoker  time  size
day                                           
Sat           87   87   87      87    87    87
Sun           76   76   76      76    76    76
Thur          62   62   62      62    62    62
Fri           19   19   19      19    19    19

In [86]:
tips.groupby("sex")["tip"].mean()

<ipython-input-86-1777249021>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("sex")["tip"].mean()


sex
Male      3.089618
Female    2.833448
Name: tip, dtype: float64

In [87]:
tips[tips["sex"]=="Female"].groupby("time").count()

<ipython-input-87-1314324474>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips[tips["sex"]=="Female"].groupby("time").count()


total_bill  tip  sex  smoker  day  size
time                                           
Lunch           35   35   35      35   35    35
Dinner          52   52   52      52   52    52

In [88]:
other=tips.groupby("size")["tip"].mean()
print(other)


size
1    1.437500
2    2.582308
3    3.393158
4    4.135405
5    4.028000
6    5.225000
Name: tip, dtype: float64


In [89]:
#smoker analysis
tips.groupby("smoker")["tip"].mean()

<ipython-input-89-3832601678>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips.groupby("smoker")["tip"].mean()


smoker
Yes    3.008710
No     2.991854
Name: tip, dtype: float64

In [90]:
#how tipping behaviour is different in male with smomking and without smoking
tips[tips.sex=="Male"].groupby("smoker")["tip"].mean()


<ipython-input-90-241982080>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips[tips.sex=="Male"].groupby("smoker")["tip"].mean()


smoker
Yes    3.051167
No     3.113402
Name: tip, dtype: float64

In [91]:
tips[tips.sex=="Female"].groupby("smoker")["tip"].mean()

<ipython-input-91-3047476729>:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips[tips.sex=="Female"].groupby("smoker")["tip"].mean()


smoker
Yes    2.931515
No     2.773519
Name: tip, dtype: float64

In [92]:
#Is there a difference in tipping behavior between male and female smokers?
tips[tips["smoker"]=="Yes"].groupby("sex")["tip"].mean()

<ipython-input-92-889446183>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  tips[tips["smoker"]=="Yes"].groupby("sex")["tip"].mean()


sex
Male      3.051167
Female    2.931515
Name: tip, dtype: float64